index.js

```diff
@@ -34,33 +37,36 @@ module.exports = function (options, cb) {
 		0x00, 0x02, /* Type:  NS      */
 		0x00, 0x01  /* Class: IN      */
 	]);
+}
 
-	var udpSocket = dgram.createSocket('udp4');
-
-	udpSocket.on('message', function (msg, rinfo) {
+function subscribeUdpSocketOnMessageReceived(udpSocket, server, options, resolve) {
```

Comment:
As you might have guessed, I'm not a fan of such long names. Why not call it `listen`?